syntax = "proto3";

message ConnectionMessage {
  message Person {
    int64 id = 1;
    string company_name = 2;
    string last_name = 3;
    string first_name = 4;
  }

  message Location {
    int64 id = 1;
    int64 person_id = 2;
    string creation_time = 3;
    string longitude = 4;
    string latitude = 5;
  }

  Person person = 1;
  Location location = 2;
}

message ConnectionRequest {
  int64 person_id = 1;
  string start_date = 2;
  string end_date = 3;
  int32 distance = 4;
}

message Empty {

}

message ConnectionMessageList {
  repeated ConnectionMessage connections = 1;
}

service ConnectionService {
  rpc Get(ConnectionRequest) returns (ConnectionMessageList);
}